### PR TITLE
Rewrite Companies&FOP autoupdater for short source file

### PR DESCRIPTION
**Pull request by PyDev777**

### CODE REVIEW

File: data_ocean/downloader.py [hunk 1]
```diff
@@ -48,44 +48,66 @@ def __init__(self):
...
-    def unzip_file(self):
+    def is_zip_file(self):
         if not zipfile.is_zipfile(self.file_path):
             msg = f'Error! {self.file_path} is not a Zip file!'
             self.log_obj.unzip_message = msg
             self.log_obj.save()
             logger.exception(f'{self.reg_name}: {msg}')
-            exit(1)
+            raise Exception('Unzip error!', msg)
+        return True
```
Comment (roman-tiukh) [inline data_ocean/downloader.py:58]: -

File: data_ocean/downloader.py [hunk 2]
```diff
@@ -48,44 +48,66 @@ def __init__(self):
...
         test_error = subprocess.run(['unzip', '-t', self.file_path]).returncode
         if test_error:
             msg = f'Error! Test archive {self.file_path} failed!'
             self.log_obj.unzip_message = msg
             self.log_obj.save()
             logger.exception(f'{self.reg_name}: {msg}')
-            exit(1)
+            raise Exception('Unzip error!', msg)
         logger.info(f'{self.reg_name}: Test archive finished successfully.')
+        return True
```
Comment (roman-tiukh) [inline data_ocean/downloader.py:70]: -

**Review by roman-tiukh — CHANGES_REQUESTED** [review]

File: business_register/converter/fop.py [hunk 3]
```diff
@@ -293,7 +295,7 @@ def get_source_file_url(self):
             return
 
         for i in r.json()['result']['resources']:
-            if self.zip_required_file_sign in i['url']:
+            if self.zip_required_file_sign not in i['url']:
```
[the remark this answers is not in the document]
Comment (PyDev777) [reply business_register/converter/fop.py]: Fixed

File: business_register/converter/company_converters/ukr_company.py [hunk 4]
```diff
@@ -750,7 +751,7 @@ def get_source_file_url(self):
             return
 
         for i in r.json()['result']['resources']:
-            if self.zip_required_file_sign in i['url']:
+            if self.zip_required_file_sign not in i['url']:
```
[the remark this answers is not in the document]
Comment (PyDev777) [reply business_register/converter/company_converters/ukr_company.py]: Fixed

**Review by roman-tiukh — APPROVED** [review]

**Pull request merged**